pages/[id]/BuildConfig.js

```diff
@@ -0,0 +1,89 @@
+import { useState } from 'react';
+import Page from '../../components/page';
+import Card, {
+    CardHeader,
+    CardTitle,
+    CardContent,
+  } from '../../components/card';
+import { HACKATHON_NAVBAR } from '../../constants';
+import {
+    db,
+    getHackathonPaths,
+    getHackathons,
+  } from '../../utility/firebase';
+  
+export default function BuildConfig({id, hackathons}) {
+    const [buildConfig, setBuildConfig] = useState({});
+
+    db.collection('Hackathons').doc(id).onSnapshot(doc => {
+        setBuildConfig(doc.data().BuildConfig)
+    })  
+
+    return(
+        <Page
+        currentPath={id}
+        hackathons={hackathons}
+        navbarItems={HACKATHON_NAVBAR}
+        >     
+            {buildConfig === null || buildConfig === undefined ? 
+                <Card>
+                    <CardHeader>
+                        <CardTitle>
+                            No BuildConfig for {id}
+                        </CardTitle>
+                    </CardHeader>                    
+                </Card>
+                :
+                [buildConfig.componentStyling === null || buildConfig.componentStyling === undefined ?
+                <Card>
+                    <CardHeader>
+                        <CardTitle>
+                            No componentStyling for {id}
+                        </CardTitle>
+                    </CardHeader>
+                </Card>
+                :
+                Object.entries(buildConfig.componentStyling).map(([key, val]) => 
+                <Card>
+                    <CardHeader>
+                        <CardTitle>{key}</CardTitle>
+                    </CardHeader>
+                    <CardContent><pre>{JSON.stringify(val, null, 2)}</pre></CardContent>
+                </Card>
+                ),
+
+                buildConfig.globalStyling === null || buildConfig.componentStyling === undefined ? 
+                <Card>
+                    <CardHeader>
+                        <CardTitle>
+                            No globalStyling for {id}
+                        </CardTitle>
+                    </CardHeader>
+                </Card>
+                :
+                <Card>
+                    <CardHeader>
+                        <CardTitle>globalStyling</CardTitle>                    
+                    </CardHeader>
+                    <CardContent><pre>{JSON.stringify(buildConfig.globalStyling, null, 2)}</pre></CardContent>
+                </Card>
+                ]
```

Comment:
There is duplicate code here, do you think you can refactor them?

Reply:
Generally, the no card/ card portions should refactor nicely. If you think it may work, maybe you could combine the globalStyling and componentStyling together into a signle array and just loop over them once?

Reply:
check out my latest commit for the refactoring! not sure if it's exactly what you had in mind but lmk